torch/_higher_order_ops/associative_scan.py

```diff
@@ -71,49 +166,110 @@ def add(x: torch.Tensor, y: torch.Tensor):
         cumsum = associative_scan(add, x, dim)
 
     """
-    assert callable(combine_fn), "combine_fn must be a callable, but got {combine_fn}"
-    assert isinstance(dim, int), "dim must be an int, but got {type(dim)}"
-
-    if not torch._dynamo.is_compiling():
-        with _set_compilation_env(), torch._dynamo.utils.disable_cache_limit():
```

Comment:
W'd better still should go through torch.compile for associative_scan to check for any side-effects and lift the free-vars.